SU2_CFD/src/solvers/CNEMONSSolver.cpp

```diff
@@ -982,36 +970,33 @@ void CNEMONSSolver::BC_Smoluchowski_Maxwell(CGeometry *geometry,
                                             CConfig *config,
                                             unsigned short val_marker) {
 
-
-  unsigned short iDim, jDim, iVar, iSpecies;
+  unsigned short iDim, iVar, iSpecies;
   unsigned short T_INDEX, TVE_INDEX, VEL_INDEX;
   unsigned long iVertex, iPoint, jPoint;
   su2double ktr, kve, Mass = 0.0;
   su2double Ti, Tvei, Tj, Tvej;
   su2double Twall, Tslip, Tslip_ve, dij;
   su2double Pi;
-  su2double Area, *Normal, UnitNormal[3];
-  su2double *Coord_i, *Coord_j;
+  su2double Area, UnitNormal[MAXNDIM];
   su2double C, alpha_V, alpha_T;
 
   su2double TMAC, TAC;
   su2double Viscosity, Eddy_Visc, Lambda;
   su2double Density, GasConstant;
 
-  su2double **Grad_PrimVar;
-  su2double Vector_Tangent_dT[3], Vector_Tangent_dTve[3], Vector_Tangent_HF[3];
+  const su2double* const* Grad_PrimVar;
```

Comment:
you could put even one more const... to say that you cannot change what is points to.
It is in my todo list to create a "matrix view" to be able to access the gradient without copying and without pointers... but I will never get there.

Reply:
If there is time in the Dev meeting tomorrow I might ask you again about this (both the code and your todo-list item) ... because i did not really get it :) 🐌